Source/AliveLibAO/Elum.cpp

```diff
@@ -1191,17 +1349,65 @@ void Elum::State_13_RunTurnToWalk_4147C0()
 
 void Elum::State_14_Speak_414860()
 {
-    NOT_IMPLEMENTED();
+    if (field_10_anim.field_92_current_frame == 3 && field_170_flags & 0x20)
+    {
+        Abe_SFX_2_42A220(21u, 75, 0, this);
+        field_170_flags &= ~0x20u;
+    }
+    else
+    {
+        field_170_flags |= 0x20u;
+    }
+
+    if (field_10_anim.field_4_flags.Get(AnimFlags::eBit18_IsLastFrame))
+    {
+        if (!ToNextState_4120F0())
+        {
+            ToIdle();
+        }
+    }
 }
 
 void Elum::State_15_Speak_4148F0()
 {
-    NOT_IMPLEMENTED();
+    if (field_10_anim.field_92_current_frame == 3 && field_170_flags & 0x20)
+    {
+        Abe_SFX_2_42A220(20u, 75, 0, this);
+        field_170_flags &= ~0x20u;
+    }
+    else
+    {
+        field_170_flags |= 0x20u;
+    }
+    if (field_10_anim.field_4_flags.Get(AnimFlags::eBit18_IsLastFrame))
+    {
+        if (!ToNextState_4120F0())
+        {
+            ToIdle();
+        }
+    }
 }
 
 void Elum::State_16_Speak_414980()
```

Comment:
this function is exactly the same as State_15_Speak_4148F0, also State_14_Speak_414860 is almost identical. You could add a helper method void Speak(int soundId); that removes the duplicated code as the first soundId passed to Abe_SFX_2_42A220() is the only change between these 3?